src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -76,61 +74,45 @@ runMiner
     => ChainwebVersion
     -> MinerResources logger cas
     -> IO ()
-runMiner v m = do
-    gen <- MWC.createSystemRandom
-    (chooseMiner gen v)
-      (logFunction $ _minerResLogger m)
+runMiner v mr = do
+    inner <- chooseMiner v
+    mining
+      inner
+      (logFunction $ _minerResLogger mr)
       conf
-      (_minerResNodeId m)
-      (_minerResCutDb m)
+      (_minerResNodeId mr)
+      (_minerResCutDb mr)
   where
     conf :: MinerConfig
-    conf = _minerResConfig m
+    conf = _minerResConfig mr
 
     miners :: MinerCount
     miners = _configTestMiners conf
 
-    chooseMiner
-        :: PayloadCas cas
-        => MWC.GenIO
-        -> ChainwebVersion
-        -> LogFunction
-        -> MinerConfig
-        -> NodeId
-        -> CutDb cas
-        -> IO ()
-    chooseMiner g Test{} = mining (localTest g miners)
-    chooseMiner g TimedConsensus{} = mining (localTest g miners)
-    chooseMiner _ PowConsensus{} = mining (localPOW v)
-    chooseMiner g TimedCPM{} = mining (localTest g miners)
-    chooseMiner _ Development = mining (localPOW v)
-    chooseMiner _ Testnet00 = mining (localPOW v)
-    chooseMiner _ Testnet01 = mining (localPOW v)
-    chooseMiner _ Testnet02 = mining (localPOW v)
-
--- | Artificially delay the mining process to simulate Proof-of-Work.
---
-localTest :: MWC.GenIO -> MinerCount -> BlockHeader -> IO BlockHeader
-localTest gen miners bh = MWC.geometric1 t gen >>= threadDelay >> pure bh
-  where
-    v :: ChainwebVersion
-    v = _blockChainwebVersion bh
-
-    t :: Double
-    t = int graphOrder / (int (_minerCount miners) * meanBlockTime * 1000000)
-
-    graphOrder :: Natural
-    graphOrder = order $ _chainGraph v
-
-    meanBlockTime :: Double
-    meanBlockTime = case blockRate v of
-        Just (BlockRate (Seconds n)) -> int n
-        Nothing -> error $ "No BlockRate available for given ChainwebVersion: " <> show v
-
--- | A single-threaded in-process Proof-of-Work mining loop.
---
-localPOW :: ChainwebVersion -> BlockHeader -> IO BlockHeader
-localPOW v = usePowHash v mine
-
--- remoteMining :: BlockHeader -> IO BlockHeader
--- remoteMining = undefined  -- TODO!
+    chooseMiner :: ChainwebVersion -> IO (BlockHeader -> IO BlockHeader)
+    chooseMiner Test{} = testMiner
+    chooseMiner TimedConsensus{} = testMiner
+    chooseMiner PowConsensus{} = powMiner
+    chooseMiner TimedCPM{} = testMiner
+    chooseMiner Development = powMiner
+    chooseMiner Testnet00 = powMiner
+    chooseMiner Testnet01 = powMiner
+    chooseMiner Testnet02 = powMiner
+
+    testMiner :: IO (BlockHeader -> IO BlockHeader)
+    testMiner = do
+        gen <- MWC.createSystemRandom
+        pure $ localTest gen miners
+
+    powMiner :: IO (BlockHeader -> IO BlockHeader)
+    powMiner = case _configRemoteMiners conf of
+        [] -> pure $ localPOW v
+        rs -> do
+            m <- newManager defaultManagerSettings
```

Comment:
Why not `tlsManagerSettings` from `http-client-tls`? It seems precedent in Chainweb to use `defaultManagerSettings`, so I went with that here.